python/cloudtik/providers/_private/_azure/config.py

```diff
@@ -1219,6 +1220,27 @@ def _configure_provision_public_ip(config):
 
     return config
 
+def check_public_ip_address_dependency(network_client, resource_group_name, public_ip_address_name):
+    check_time = RESOURCE_CHECK_TIME
+    cli_logger.print(
+        "Checking the  dependency of public ip address...")
+    while check_time > 0:
+        public_ip_address = network_client.public_ip_addresses.get(
+            resource_group_name=resource_group_name,
+            public_ip_address_name=public_ip_address_name)
+        if public_ip_address.ip_configuration is not None:
+            if check_time > 0:
+                check_time = check_time - 1
+                cli_logger.warning("The dependency of public ip address still exists. "
+                                   "Remaining {} tries to check the dependency...".format(check_time))
+                time.sleep(10)
```

Comment:
Do you have experiences usually how long for the the dependency will be released?
Because this is in the very key code path of terminate_node, we need be very careful about the time to sleep to avoid performance issues. (The terminate node method will also be called in the cluster_controller)

So we should use short sleep time but we can tune up the MAX NUMBER OF RETRIES.